app/controllers/exchanges/employer_applications_controller.rb

```diff
@@ -1,72 +1,88 @@
+      if result
+        flash[:notice] = "#{@benefit_sponsorship.organization.legal_name}'s Application canceled successfully."
+      else
+        flash[:error] = "#{@benefit_sponsorship.organization.legal_name}'s Application could not be canceled due to #{errors.inject(''){|_memo, _error| "#{memo}<li>#{error}</li>"}.html_safe}"
+      end
+      render :js => "window.location = #{exchanges_hbx_profiles_root_path.to_json}"
     end
-    render json: @reasons
-  end
 
-  def reinstate
+    def term_reasons
+      @reasons = if params[:reason_type_id] == "term_actions_nonpayment"
+                   BenefitSponsors::BenefitApplications::BenefitApplication::NON_PAYMENT_TERM_REASONS
+                 else
+                   BenefitSponsors::BenefitApplications::BenefitApplication::VOLUNTARY_TERM_REASONS
+                 end
+      render json: @reasons
+    end
 
-  end
+    def reinstate
+      if ::EnrollRegistry.feature_enabled?(:benefit_application_reinstate)
+        application = @benefit_sponsorship.benefit_applications.find(params[:employer_application_id])
+        transmit_to_carrier = params['transmit_to_carrier'] == "true"
+        result = EnrollRegistry[:benefit_application_reinstate]{ {params: {benefit_application: application, options: {transmit_to_carrier: transmit_to_carrier} } } }
+        if result.success?
+          flash[:notice] = "#{application.benefit_sponsorship.legal_name} - #{l10n('exchange.employer_applications.success_message')} #{(application.canceled? ? application.start_on : application.end_on.next_day).to_date}"
+        else
+          flash[:error] = "#{application.benefit_sponsorship.legal_name} - #{result.failure}"
+        end
+      end
+      redirect_to exchanges_hbx_profiles_root_path
+    rescue StandardError => e
+      Rails.logger.error { "#{application.benefit_sponsorship.legal_name} - #{l10n('exchange.employer_applications.unable_to_reinstate')} - #{e.backtrace}" }
+      redirect_to exchanges_hbx_profiles_root_path, flash[:error] => "#{application.benefit_sponsorship.legal_name} - #{l10n('exchange.employer_applications.unable_to_reinstate')}"
+    end
 
-  private
+    private
 
-  def can_modify_plan_year?
-    authorize HbxProfile, :can_modify_plan_year?
-  end
+    def can_modify_plan_year?
+      authorize HbxProfile, :can_modify_plan_year?
+    end
 
-  def check_hbx_staff_role
-    unless current_user.has_hbx_staff_role?
-      redirect_to root_path, :flash => { :error => "You must be an HBX staff member" }
+    def check_hbx_staff_role
+      redirect_to root_path, :flash => { :error => "You must be an HBX staff member" } unless current_user.has_hbx_staff_role?
     end
-  end
 
-  def find_benefit_sponsorship
-    @benefit_sponsorship = BenefitSponsors::BenefitSponsorships::BenefitSponsorship.find(params[:employer_id])
+    def find_benefit_sponsorship
+      @benefit_sponsorship = BenefitSponsors::BenefitSponsorships::BenefitSponsorship.find(params[:employer_id])
+    end
   end
 end
```

Comment:
@nks2109  - Any reason why you ran rubocop auto fixes on this file ?

It makes sense to run rubocop on entire file only if you're going to re-write / update most of the file contents;

Reply:
As I was getting rubocop issues for this file, we will refactor this controller as this related to terminate, cancel, reinstate, and revise end date actions controller for this reinstatement project so I have updated in this PR.